File: HW7/151044027_CSE241_HW7/src/PolygonVect.java
import java.awt.Graphics;
import java.util.ArrayList;



public class PolygonVect extends Polygon{
	public class Point2D
	{
		private int xCoordinate;
		private int yCoordinate;
		Point2D(int theXCoordinate, int theYCoordinate)
		{
			setTwoDx(theXCoordinate);
			setTwoDy(theYCoordinate);
		}
		public int getTwoDx() {
			return xCoordinate;
		}
		public void setTwoDx(int xCoordinate) {
			this.xCoordinate = xCoordinate;
		}
		public int getTwoDy() {
			return yCoordinate;
		}
		public void setTwoDy(int yCoordinate) {
			this.yCoordinate = yCoordinate;
		}
	}
	private ArrayList<Point2D> connectedDots;
	public PolygonVect()
	{
	}
	public PolygonVect(int theCapacity)
	{
		connectedDots = new ArrayList<Point2D>();
	}
	public PolygonVect(Rectangle rectangleObject)
	{
		connectedDots = new ArrayList<Point2D>();
		int x1,y1;
		int x2,y2;
		int x3,y3;
		int x4,y4;
		int width;
		int height;

		width = (int)rectangleObject.getWidth();
		height = (int)rectangleObject.getHeight();
		x1 = (int)rectangleObject.getXCoordinate();
		y1 = (int)rectangleObject.getYCoordinate();


		x2 = x1;
		y2 = y1 + height;
		x3 = x1 + width;
		y3 = y1 + height;
		x4 = x1 + width;
		y4 = y1;


		connectedDots.get(0).setTwoDx(x1);
		connectedDots.get(0).setTwoDy(y1);

		connectedDots.get(1).setTwoDx(x2);
		connectedDots.get(1).setTwoDy(y2);

		connectedDots.get(2).setTwoDx(x3);
		connectedDots.get(2).setTwoDy(y3);

		connectedDots.get(3).setTwoDx(x4);
		connectedDots.get(3).setTwoDy(y4);

	}
	
	public PolygonVect(Triangle triangleObject)
	{
		int x1,y1,x2,y2,x3,y3;
		int width;

		connectedDots = new ArrayList<Point2D>();
		width = (int)triangleObject.getLength();
		x1 = (int)triangleObject.getX1Coordinate();
		y1 = (int)triangleObject.getY1Coordinate();
		x2 = (int)triangleObject.getX2Coordinate();
		y2 = (int)triangleObject.getY2Coordinate();
		x3 = (int)triangleObject.getX3Coordinate();
		y3 = (int)triangleObject.getY3Coordinate();

		connectedDots.get(0).setTwoDx(x1);
		connectedDots.get(0).setTwoDy(y1);

		connectedDots.get(1).setTwoDx(x2);
		connectedDots.get(1).setTwoDy(y2);

		connectedDots.get(2).setTwoDx(x3);
		connectedDots.get(2).setTwoDy(y3);


	}
	
	public PolygonVect(Circle circleObject)
	{
		
		double counterDegree = 0;
		
		connectedDots = new ArrayList<Point2D>();
		int x,y,radius;
		double tempx, tempy;
		x = (int)circleObject.getXCoordinate();
		y = (int)circleObject.getYCoordinate();
		radius = (int)circleObject.getRadius();
		for(int i = 0 ; i < 100 ; ++i)
		{
			counterDegree += 3.6;
			tempx = x + radius * Math.cos(counterDegree * 3.14 / 180);
			tempy = y + radius * Math.sin(counterDegree * 3.14 / 180);
			connectedDots.get(i).setTwoDx((int)tempx);
			connectedDots.get(i).setTwoDy((int)tempy);
			
		}
	}
	@Override
	public double area() {
		// You can't calculate unknown area.
		return -1;
	}

	@Override
	public double perimeter() {
		// You can't calculate unknown perimeter
		return -1;
	}

	@Override
	public void increment() {
		for(int i = 0 ; i < connectedDots.size() ; ++i)
		{
			connectedDots.get(i).setTwoDx(connectedDots.get(i).getTwoDx() + 1);
			connectedDots.get(i).setTwoDy(connectedDots.get(i).getTwoDy() + 1);
		}
		
	}

	@Override
	public void decrement() {
		for(int i = 0 ; i < connectedDots.size() ; ++i)
		{
			connectedDots.get(i).setTwoDx(connectedDots.get(i).getTwoDx() - 1);
			connectedDots.get(i).setTwoDy(connectedDots.get(i).getTwoDy() - 1);
		}
		
	}

	@Override
	public void draw(Graphics graph) {
		// TODO Auto-generated method stub
		int polygonX [] = new int [connectedDots.size()];
		int polygonY [] = new int [connectedDots.size()];
		for(int i = 0 ; i < connectedDots.size() ; ++i)
		{
			polygonX[i] = connectedDots.get(i).getTwoDx();
			polygonY[i] = connectedDots.get(i).getTwoDy();
		}
		graph.drawPolygon(polygonX, polygonY, connectedDots.size());
	}
	@Override
	public int toCompare(Shape genericShape) {
		// TODO Auto-generated method stub
		return 0;
	}
}

File: HW5/151044027_CSE241_HW5/PolygonVect.cpp
#include "PolygonVect.h"

namespace PatogluShape
{
	PolygonVect::PolygonVect()
	{
		STLDots.resize(100);
	}
	//default constructor.
	PolygonVect::PolygonVect(const vector<Point2D> &twoDPoints)
	{
		STLDots.resize(twoDPoints.size());
		for(auto i = 0 ; i < twoDPoints.size() ; ++i)
		{
			STLDots[i].setTwoDx(twoDPoints[i].getTwoDx());
			STLDots[i].setTwoDy(twoDPoints[i].getTwoDy());
		}
	}
	PolygonVect::PolygonVect(const Point2D &point)
	{
		STLDots.resize(1);
		STLDots[0].setTwoDx(point.getTwoDx());
		STLDots[0].setTwoDy(point.getTwoDy());
	}

	PolygonVect::PolygonVect(const Rectangle& rectangleObject)
	{
		STLDots.resize(4);
		double x1,y1;
		double x2,y2;
		double x3,y3;
		double x4,y4;
		double width;
		double height;
		width = rectangleObject.getWidth();
		height = rectangleObject.getHeight();
		x1 = rectangleObject.getXCoordinate();
		y1 = rectangleObject.getYCoordinate();
		x2 = x1;
		y2 = y1 + height;
		x3 = x1 + width;
		y3 = y1 + height;
		x4 = x1 + width;
		y4 = y1;

		STLDots[0].setTwoDx(x1);
		STLDots[0].setTwoDy(y1);

		STLDots[1].setTwoDx(x2);
		STLDots[1].setTwoDy(y2);

		STLDots[2].setTwoDx(x3);
		STLDots[2].setTwoDy(y3);

		STLDots[3].setTwoDx(x4);
		STLDots[3].setTwoDy(y4);

	}
	PolygonVect::PolygonVect(const Triangle& triangleObject)
	{
		double x1,y1,x2,y2,x3,y3;
		double width;
		STLDots.resize(3);
		width = triangleObject.getWidth();
		x1 = triangleObject.getX1Coordinate();
		y1 = triangleObject.getY1Coordinate();
		x2 = triangleObject.getX2Coordinate();
		y2 = triangleObject.getY2Coordinate();
		x3 = triangleObject.getX3Coordinate();
		y3 = triangleObject.getY3Coordinate();

		STLDots[0].setTwoDx(x1);
		STLDots[0].setTwoDy(y1);

		STLDots[1].setTwoDx(x2);
		STLDots[1].setTwoDy(y2);

		STLDots[2].setTwoDx(x3);
		STLDots[2].setTwoDy(y3);
	}
	PolygonVect::PolygonVect(const Circle& circleObject)
	{
		double counterDegree = 0;
		STLDots.resize(100);
		double x,y,radius;
		double tempx, tempy;
		x = circleObject.getXCoordinate();
		y = circleObject.getYCoordinate();
		radius = circleObject.getRadius();
		for(auto i = 0 ; i < 100 ; ++i)
		{
			counterDegree += 3.6;
			tempx = x + radius * cos(counterDegree * 3.14 / 180);
			tempy = y + radius * sin(counterDegree * 3.14 / 180);
			STLDots[i].setTwoDx(tempx);
			STLDots[i].setTwoDy(tempy);
			
		}		
	}

	PolygonVect::PolygonVect(int theCapacity)
	{
		STLDots.resize(theCapacity);
	} //Polygon that creates an array with dynamic input.
	PolygonVect::Point2D& PolygonVect::operator[](int theIndex)
	{
		return STLDots[theIndex];
	}
	bool PolygonVect::operator==(const PolygonVect& rtSide) const
	{
		bool isEqual = true;
		for(auto i = 0 ; i < STLDots.size() ; ++i)
		{
			if((STLDots[i].getTwoDx() == rtSide.STLDots[i].getTwoDx()) && (STLDots[i].getTwoDy() == rtSide.STLDots[i].getTwoDy()))
			{
				//very good.
			}
			else
				isEqual = false;
		}
		return isEqual;
	}
	bool PolygonVect::operator!=(const PolygonVect& rtSide) const
	{
		bool isEqual = true;
		for(auto i = 0 ; i < STLDots.size() ; ++i)
		{
			if((STLDots[i].getTwoDx() == rtSide.STLDots[i].getTwoDx()) && (STLDots[i].getTwoDy() == rtSide.STLDots[i].getTwoDy()))
			{
				//very good.
			}
			else
				isEqual = false;
		}
		return !(isEqual);
	}

	const PolygonVect PolygonVect::operator+(const PolygonVect& rtSide) const
	{
		int i;
		int rtSideCapacity;
		int totalCapacity;
		rtSideCapacity = rtSide.STLDots.size();
		totalCapacity = STLDots.size() + rtSideCapacity;
		PolygonVect bigPolygon(totalCapacity);
	

		for(i = 0 ; i < STLDots.size(); ++i)
		{
			bigPolygon.STLDots[i] = STLDots[i];
		}
		for(auto j = i ; j < bigPolygon.STLDots.size() ; ++j)
		{
			i = 0;
			i++;
			bigPolygon.STLDots[j] = rtSide.STLDots[i];
		}

		
		return bigPolygon;
	}
	ostream& operator <<(ostream& output, const PolygonVect& rtSide)
	{
		//<polygon points="200,10 250,190 160,210" style="fill:lime;stroke:purple;stroke-width:1" />
		output << "<polygon points =\"";
		for(auto i = 0; i < rtSide.STLDots.size() ; ++i)
		{
			//output << "rtUsed " << rtSide.used << endl; 
			//output << " " << rtSide.connectedDots->getTwoDx() << ',' << rtSide.connectedDots->getTwoDy();
			if(i < rtSide.STLDots.size() - 1)
				output << " " << rtSide.STLDots[i].getTwoDx() << ',' << rtSide.STLDots[i].getTwoDy() << ',';
			else
				output << " " << rtSide.STLDots[i].getTwoDx() << ',' << rtSide.STLDots[i].getTwoDy();
		}
		//output << " " << rtSide.connectedDots[1].getTwoDx() << ',' << rtSide.connectedDots[1].getTwoDy();
		output << "\"";
		output << " style=\"fill:lime;stroke:purple;stroke-width:1\" />";
		return output;

	}
	PolygonVect& PolygonVect::operator=(const PolygonVect& rightSide)
	{
		if(STLDots.size()!= rightSide.STLDots.size())
		{
			STLDots.clear();
			
			STLDots.resize(rightSide.STLDots.size());
		}
		
			for(auto i = 0; i <rightSide.STLDots.size() ; ++i)
			{

				STLDots[i].setTwoDx(rightSide.STLDots[i].getTwoDx());
				STLDots[i].setTwoDy(rightSide.STLDots[i].getTwoDy());
			}
		
		return *this;
	}

	Shape* PolygonVect:: operator++() 
	{
		Shape *a;
		cerr << "You shouldn't increment Polygon objects." << endl;
		return a;
	}
	Shape* PolygonVect:: operator++(int) 
	{
		Shape *a;
		cerr << "You shouldn't increment Polygon objects." << endl;
		return a;
	}
	Shape* PolygonVect:: operator--() 
	{
		Shape *a;
		cerr << "You shouldn't increment Polygon objects." << endl;
		return a;
	}
	Shape* PolygonVect::operator--(int) 
	{
		Shape *a;
		cerr << "You shouldn't increment Polygon objects." << endl;
		return a;
	}


}